### ISSUE TITLE
Errors when there is "lambda" in the python file

### ISSUE THREAD

**Issue by matthew-z** [proofit404/anaconda-mode#138]
Hi

When I write `lambda` the emacs gives me an error, the backtraces are:

Debugger entered--Lisp error: (wrong-type-argument stringp ((data . "KeyError(u'pairs',)") (code . -32000) (message . "Server error")))
  format(((data . "KeyError(u'pairs',)") (code . -32000) (message . "Server error")))
  apply(format ((data . "KeyError(u'pairs',)") (code . -32000) (message . "Server error")))
  error(((data . "KeyError(u'pairs',)") (code . -32000) (message . "Server error")))
  #[257 "p\306\307\310\311\312\313^F^F!\314\"\315\316%DC\216\304\317 \232\2036^@r\320\304!q\210\303p\232?\2062^@\302`\232?\2062^@\305\321 \232?)\203=^@\322\3$
  apply(#[257 "p\306\307\310\311\312\313^F^F!\314\"\315\316%DC\216\304\317 \232\2036^@r\320\304!q\210\303p\232?\2062^@\302`\232?\2062^@\305\321 \232?)\203=^@$
  url-http-activate-callback()
  url-http-content-length-after-change-function(127 240 113)
  url-http-wait-for-headers-change-function(1 245 244)
  url-http-generic-filter(#<process 127.0.0.1> "HTTP/1.1 500 Internal Server Error^M\nServer: BaseHTTP/0.3 Python/2.7.10^M\nDate: Mon, 09 Nov 2015 07:50:57 G$

The info in anaconda-mode is 

anaconda_mode port 9000
127.0.0.1 - - [09/Nov/2015 16:50:57] ================================================================================
Traceback (most recent call last):
  File "/home/matt/.emacs.d/.cache/anaconda-mode/0.1.1/service_factory/providers/basehttp.py", line 40, in do_POST
    status, response = self.server.service(data)
  File "/home/matt/.emacs.d/.cache/anaconda-mode/0.1.1/service_factory/service.py", line 49, in __call__
    result = self.apply(method, args)
  File "/home/matt/.emacs.d/.cache/anaconda-mode/0.1.1/service_factory/service.py", line 104, in apply
    server_error(args['id'], error)
  File "/home/matt/.emacs.d/.cache/anaconda-mode/0.1.1/service_factory/errors.py", line 92, in server_error
    six.raise_from(ServiceException(500, dumps(response)), error)
  File "/home/matt/.emacs.d/.cache/anaconda-mode/0.1.1/six.py", line 718, in raise_from
    raise value
ServiceException: (500, '{"jsonrpc": "2.0", "id": 1, "error": {"message": "Server error", "code": -32000, "data": "KeyError(u\'pairs\',)"}}')

python version 2.7.10 :: anaconda 2.3.0
Emacs 24.5.1 Spacemacs 0.104.4

This problem happens in my laptop and Ubuntu Server both.

Any ideas?
Thanks

**Comment by proofit404** [comment]
This is a jedi library bug. I can't fix it in the anaconda-mode.

**Comment by matthew-z** [comment]
Thank you for the advice.

I thought that it is the problem caused by JEDI 0.9 but I checked the jedi version use `pip list` and the result was that the jedi is `0.8.1`. So i did not run the command which shows on the homepage of the repo.

After receiving your message, I tried to delete the jedi in the anaconda sever and reinstall 0.8.1 follow your instruction

M-: (dired (anaconda-mode-server-directory)) RET
M-! rm -rf jedi* RET
M-! pip install "jedi<0.9" -t . RET


Then the problem solved.
